packages/ember-runtime/lib/mixins/array.js

```diff
@@ -517,18 +518,22 @@ export default Mixin.create(Enumerable, {
 
     sendEvent(this, '@array:change', [this, startIdx, removeAmt, addAmt]);
 
+    // firstObject/lastObject change notifications
+    // The call to objectAt() should only occur if there is a cached value
     var length = get(this, 'length');
-    var cachedFirst = cacheFor(this, 'firstObject');
-    var cachedLast = cacheFor(this, 'lastObject');
-
-    if (objectAt(this, 0) !== cachedFirst) {
-      propertyWillChange(this, 'firstObject');
-      propertyDidChange(this, 'firstObject');
-    }
-
-    if (objectAt(this, length - 1) !== cachedLast) {
-      propertyWillChange(this, 'lastObject');
-      propertyDidChange(this, 'lastObject');
+    var meta = peekMeta(this);
+    var cache  = meta && meta.readableCache();
+    if (cache) {
+        if (cache['firstObject'] !== undefined &&
```

Comment:
We use 2 space indentation, the 4 spaces here (and below) are causing the build failures).